core/src/main/scala/org/apache/spark/util/ClosureCleaner.scala

```diff
@@ -109,7 +109,16 @@ private[spark] object ClosureCleaner extends Logging {
 
   private def createNullValue(cls: Class[_]): AnyRef = {
     if (cls.isPrimitive) {
-      new java.lang.Byte(0: Byte) // Should be convertible to any primitive type
+      if (cls == java.lang.Boolean.TYPE) {
```

Comment:
Can you use `match` syntax here?
I see why `boolean` needs a special case, but a `byte` widens to a `char` in Java so I don't think that's necessary?